KubeArmor/types/types.go

```diff
@@ -424,6 +425,45 @@ type CapabilitiesType struct {
 	Action   string   `json:"action,omitempty"`
 }
 
+// SyscallFromSourceType Structure
+type SyscallFromSourceType struct {
+	Path      string `json:"path,omitempty"`
+	Dir       string `json:"dir,omitempty"`
+	Recursive bool   `json:"recursive,omitempty"`
+}
+
+// SyscallMatchType Structure
+type SyscallMatchType struct {
+	Syscalls   []string                `json:"syscall,omitempty"`
+	FromSource []SyscallFromSourceType `json:"fromSource,omitempty"`
+
+	Severity int      `json:"severity,omitempty"`
+	Tags     []string `json:"tags,omitempty"`
+	Message  string   `json:"message,omitempty"`
+}
+
+// SyscallMatchPathType Structure
+type SyscallMatchPathType struct {
+	Path       string                  `json:"path,omitempty"`
+	Recursive  bool                    `json:"recursive,omitempty"`
```

Comment:
We using Path as both path and directory now?

Reply:
Yes, we merged matchpaths and matchdirectories in syscalls for now, we are going to merge them for all other types in the upcoming releases.